frontend/dispatcher.lua

```diff
@@ -254,6 +244,40 @@ function Dispatcher:addSubMenu(menu, location, settings)
                 })
             end
         end
+    end
+end
+
+--[[--
+Add a submenu to edit which items are dispatched
+arguments are:
+    1) self
+    2) the table representing the submenu (can be empty)
+    3) the name of the parent of the settings table (must be a child of self)
+    4) the name of the settings table
+example usage:
+    Dispatcher.addSubMenu(self, sub_items, "profiles", "profile1")
+--]]--
+function Dispatcher:addSubMenu(menu, location, settings)
+    if not Dispatcher.initialized then Dispatcher:init() end
+    table.insert(menu, {
+        text = _("None"),
+        separator = true,
+        checked_func = function()
+            return next(self[location][settings]) == nil
+        end,
+        callback = function(touchmenu_instance)
+            self[location][settings] = {}
+            if touchmenu_instance then touchmenu_instance:updateItems() end
+        end,
+    })
+    for _,section in ipairs({"Device", "Crengine",}) do
+        local submenu = {}
+         Dispatcher.addItem(self, submenu, location, settings, section)
+        table.insert(menu, {
+            text = section,
```

Comment:
Btw, I'm not sure this is necessarily the best idea (e.g., is crengine transliterated in different scripts?), although for advanced settings it probably doesn't really matter. I do however dislike looking at that uppercase `Crengine=true`.

Reply:
Or name these sections "Rolling" and "Paging" (we have a single "rolling" engine: crengine - but multiple "paging" engines: mupdf, djvu).
or RollingEngine, or isRolling, or else

Reply:
I changed to using a k/v table for separate key and text. This should fix this issue

For the display text most users probably are not aware of rolling vs paging (crengine also has pages).
but maybe i should use epub and pdf? as those are the most obvious to the user.
please let me know of any better naming strategy :)

Reply:
Something like dynamic and fixed layout?

Reply:
let me explain better, the sections are used as submenus to divide the actions according to when they apply.

Device is always, Crengine (or whatever we name it) when using crengine etc. some items may be both in crengine and mupdf.
for example: font size is in both but calls a different event for each, to the user they would appear to be the same `Font Size` so we need different sections to let the user know which one to chose.

Therefore the string has to convey to the user which engine is being used something that the user does not necessarily pay attention to.

screenshot:
![Reader_2020-07-09_140707](https://user-images.githubusercontent.com/38916402/87075464-62ab6b80-c1ee-11ea-826a-3ac8694b7526.png)

Reply:
`Reflowable documents` (epub, fb2, txt)
`Fixed layout documents` (pdf, djvu, pics)
(except that we have the option to "Reflow" PDF, making them reflowable :/)

So, that means you should have a profile for things crengine, and another for things pdf? Can't have one with both kind of settings (that would apply as no-op if the document if from the other kind) ?

Reply:
> So, that means you should have a profile for things crengine, and another for things pdf? Can't have one with both kind of settings (that would apply as no-op if the document if from the other kind) ?

You can have both in one profile (I just did not yet implement mupdf events yet in the screenshot), and it will be a no-op, the sections are so the user is aware which ones will apply to a given document.

also some may be checked in both `GoToBeginning` is the same event for both so one action can be in both submenus, as it is reader specific